modin/pandas/indexing.py

```diff
@@ -551,20 +642,21 @@ def __getitem__(self, key):
                 result_slice = self.df.columns.slice_locs(col_loc.start, col_loc.stop)
                 return self.df.iloc[:, slice(*result_slice)]
 
+        if is_boolean_array(row_loc) and isinstance(row_loc, Series):
+            return self._handle_boolean_masking(row_loc, col_loc)
+
         row_lookup, col_lookup = self._compute_lookup(row_loc, col_loc)
-        if any(i == -1 for i in row_lookup) or any(i == -1 for i in col_lookup):
-            raise KeyError(
-                "Passing list-likes to .loc or [] with any missing labels is no longer "
-                "supported, see https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"
-            )
```

Comment:
this check was moved to the [`_compute_lookup`](https://github.com/dchigarev/modin/blob/44e747e2504a0140a6d5fc206276c3b5ba6e3402/modin/pandas/indexing.py#L830-L841)